src/main/java/UseCases/CookbookUseCase.java

```diff
@@ -27,11 +26,11 @@ public CookbookUseCase(RecipeStorage recipes, TagStorage tags) {
      * @param command
      * @return RecipeResponseImpl containing a list of recipes found
      */
-    public RecipeResponseImpl run(CommandImpl command) {
```

Comment:
This is wrong. You still haven't extended the use-case in the class. I checked the previous PR and I had marked this as an issue but you resolved it without actually fixing it. Implement the Use case interface, then this should produce an error because we DONT return implementations, only interfaces.

Reply:
Maybe the interface is wrong and needs to return the non-impl.

Reply:
@DerekCresswell my bad on forgetting to implement the interfaces.. 

How do i just return the interface? I don't understand because don't I have to physically add that list into a response?

Reply:
Like I cant add the list into the interface without it being the impl.

Reply:
You are confusing the return type of the function and the actual value you return. The function should return a Response interface. The internals of the function will create an impl and return it because it is a subclass of the interface.